readdy/main/common/Timer.cpp

```diff
@@ -21,23 +21,161 @@
 
 
 /**
- * << detailed description >>
- *
  * @file Timer.cpp
- * @brief << brief description >>
+ * @brief Implementation of timer classes
  * @author chrisfroe
  * @date 01.09.17
  * @copyright GNU Lesser General Public License v3.0
  */
 
 #include <readdy/common/Timer.h>
+#include <readdy/common/string.h>
 
 namespace readdy {
 namespace util {
 
-Timer::cumulative_time_map Timer::cumulativeTime {};
-Timer::counts_map Timer::_counts {};
-std::mutex Timer::mutex {};
+constexpr const char PerformanceNode::slash[];
+
+PerformanceNode::PerformanceNode(const std::string &name, bool measure)
+        : _name(validateName(name)), _measure(measure), _data(0., 0), _root(std::cref(*this)) { }
+
+PerformanceNode::PerformanceNode(const std::string &name, bool measure, const PerformanceNode& root)
+        : _name(validateName(name)), _measure(measure), _data(0., 0), _root(root) { }
+
+const PerformanceNode &PerformanceNode::subnode(const std::string &name) const {
+    auto validatedName = validateName(name);
```

Comment:
I think it makes sense to trim the name or forbid names with leading/trailing whitespaces, as you are trimming it in parse as well